core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -178,7 +181,9 @@ private TableMetadata getTableMetadata(Scan scan) throws ExecutionException {
   }
 
   public Optional<List<Key>> get(Scan scan) {
-    if (isWriteSetOverlappedWith(scan)) {
+    if (!ScalarDbUtils.isRelational(scan) && isWriteSetOverlappedWith(scan)) {
+      // Verify overlaps only for non-relational scan since we do it later for relational scan
+      // (right before returning the results).
       throw new IllegalArgumentException("reading already written data is not allowed");
     }
```

Comment:
Since it's moved to `verify()`, we can delete this, right?

Reply:
Sorry for that. I removed it (and fix some related tests) in 2bdbae4.